usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,832 @@
+                    return;
+                }
+            }
+            desktopActionInProgress = true;
+            if (event.isLeftButton) {
+                var distanceToBoard = Vec3.distance(whiteBoardIntersectionData.intersection, MyAvatar.position);
+                var paintSphereDimensions = Entities.getEntityProperties(_this.entityID, 'dimensions').dimensions;
+                previousStrokeWidth = paintSphereDimensions.x;
+                if (whiteBoardIntersectionData.intersects && distanceToBoard <= DRAW_ON_BOARD_DISTANCE_DESKTOP_M) { // draw on board
+                    var currentWhiteboard = whiteBoardIntersectionData.entityID;
+                    var whiteboardProperties = Entities.getEntityProperties(currentWhiteboard, ['position', 'rotation']);
+                    previousNormal = Vec3.multiply(-1, Quat.getFront(whiteboardProperties.rotation));
+                    lineStartPosition = Vec3.subtract(whiteBoardIntersectionData.intersection, Vec3.multiply(previousNormal, 
+                        STROKE_FORWARD_OFFSET_M));
+                    wasLastPointOnBoard = true;
+                } else { // draw in air
+                    lineStartPosition = Vec3.sum(pickRay.origin, Vec3.multiply(pickRay.direction, 
+                        DISTANCE_TO_DRAW_IN_FRONT_OF_CAMERA_DESKTOP_M));
+                    previousNormal = DEFAULT_NORMAL;
+                    wasLastPointOnBoard = false;
+                }
+                _this.playSound(DRAW_SOUND, DRAW_SOUND_VOLUME, lineStartPosition, true, true);
+                currentLinePoint = lineStartPosition;
+            } else if (event.isMiddleButton) {
+                _this.deleteOnBoard(event, whiteBoardIntersectionData);
+            }
+        }, 
+
+        /* ON MOUSE MOVE: Calculate the next line point and add it to the entity. If there are too many line points, 
+    begin a new line. */
+        mouseContinueLine: function(event) {
+            if (tablet.tabletShown) {
+                return;
+            }
+            if (!desktopActionInProgress) {
+                return;
+            }
+            previousLinePoint = currentLinePoint;
+            pickRay = Camera.computePickRay(event.x, event.y);
+            var whiteBoardIntersectionData = Entities.findRayIntersection(pickRay, true, whiteboardParts);
+            if (event.isLeftButton) {
+                if (whiteBoardIntersectionData.intersects) {
+                    var intersectedWhiteboardPartName = Entities.getEntityProperties(whiteBoardIntersectionData.entityID, 
+                        'name').name;
+                    // print("INTERSECTED ", intersectedWhiteboardPartName);
+                    if (intersectedWhiteboardPartName !== "Whiteboard") {
+                        _this.stopDrawing();
+                        return;
+                    }
+                }
+                var distanceToBoard = Vec3.distance(whiteBoardIntersectionData.intersection, MyAvatar.position);
```

Comment:
do you want this distance check between the _intersection point_ and the avatar position, or the _whiteboard entity position_ and the avatar position? what happens if the whiteboard is large?

Reply:
I think I want the distance to the intersection as I'm checking that the avatar is close enough to actually be trying to draw on the whiteboard. I may not be understanding what you mean...maybe we need to discuss in world?

Reply:
i think my question is more of an user experience one - so let's punt on this until someone complains (maybe nobody ever will)